src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -749,19 +749,22 @@ private static string FindProgramInPath(string program)
             return null;
         }
 
+        private static Lazy<int> _ticksPerSecond = new Lazy<int>(() => {
+            // Look up the number of ticks per second in the system's configuration
+            return Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_CLK_TCK);
+        });
+
         /// <summary>Convert a number of "jiffies", or ticks, to a TimeSpan.</summary>
         /// <param name="ticks">The number of ticks.</param>
         /// <returns>The equivalent TimeSpan.</returns>
         internal static TimeSpan TicksToTimeSpan(double ticks)
         {
-            // Look up the number of ticks per second in the system's configuration,
-            // then use that to convert to a TimeSpan
-            long ticksPerSecond = Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_CLK_TCK);
-            if (ticksPerSecond <= 0)
+            if (_ticksPerSecond.Value <= 0)
```

Comment:
It'lll be pretty common for programs to use Process but never access one of the properties that uses this, so it's a little unfortunate to pay for the statics associated with the lazy instance.  Also since we're dealing with an Int32, we don't need all the safety `Lazy<T>` provides.  How about instead just doing:
```C#
private static volatile int s_ticksPerSecond;
...
int ticksPerSecond = s_ticksPerSecond;
if (ticksPerSecond == 0)
{
    // Look up the number of ticks per second in the system's configuration,
    // then use that to convert to a TimeSpan
    ticksPerSecond = Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_CLK_TCK);
    if (ticksPerSecond <= 0)
    {
        throw new Win32Exception();
    }
    s_ticksPerSecond = ticksPerSecond;
}
return TimeSpan.FromSeconds(ticks / (double)ticksPerSecond);
```

Reply:
Pushed a fix.
Thanks!

Reply:
@stephentoub , so I understand, why is the `volatile` needed? Why not just read into a local, verify the local is `> 0`, then write it out into a regular static? 

Is the concern is that a thread would somehow read non-zero, then read again and get zero?

Reply:
Technically on a weak memory model a read could be introduced such that the code read s_ticksPerSecond twice, comparing the second to 0 and seeing that it's not, but then returning the first which may still be.  In practice that'll probably never happen, but adding the volatile read is good hygiene.